### PR TITLE
Keep imgix props

### PR DESCRIPTION
**Pull request by adebroux**
## Problem
Makers cannot use their own imgix props anymore by attaching them to the URL.

## Solution
Keep imgix props passed in through the URL.

### CODE REVIEW

**Comment by adebroux** [comment]
[Appropriate Image Sizing - avoid overwriting maker-supplied Imgix parameters with our own](https://trello.com/c/S0dP84m2/113-appropriate-image-sizing-avoid-overwriting-maker-supplied-imgix-parameters-with-our-own)

File: src/lib/utils.js [hunk 1]
```diff
@@ -15,20 +15,36 @@ export const applyImgixParameters = (source, layout, imgixProps = {}) => {
...
   const params = {
     w: layout.width,
     h: layout.height,
     dpr: Math.max(PixelRatio.get(), 2),
     auto: Platform.OS === 'web' ? 'format,compress' : 'compress',
     ...(typeof imgixProps === 'object' && { ...imgixProps }),
+    ...queryValues,
   }
-  const queryParams = Object.keys(params).map(key => `${key}=${params[key]}`).join('&')
+  const queryParams = Object.keys(params)
```
Comment (guislain-adalo) [inline src/lib/utils.js:40]: it already exists in the current code but query params need to be encoded when added to a URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

File: src/lib/utils.js [hunk 2]
```diff
@@ -15,20 +15,36 @@ export const applyImgixParameters = (source, layout, imgixProps = {}) => {
     return source
   }
 
-  const path = uri.split(/[?#]/)[0]
+  const [path, query] = uri.split(/[?#]/)
```
[the remark this answers is not in the document]
Comment (adebroux) [reply src/lib/utils.js:18]: It looks like the URL class isn't implemented in React Native, so this wouldn't work for native builds: https://www.davidangulo.xyz/posts/use-url-class-in-react-native/

[the remark this answers is not in the document]
Comment (guislain-adalo) [reply src/lib/utils.js:18]: 😢

[the remark this answers is not in the document]
Comment (michael-adalo) [reply src/lib/utils.js:18]: Just for completeness, we could try the polyfill mentioned in the blog post:
https://github.com/AdaloHQ/runner/pull/742#discussion_r1246146323

**Review by michael-adalo — APPROVED** [review]
LGTM!

**Pull request merged**